spiffe-helper.go

```diff
@@ -96,72 +124,83 @@ func (s *Sidecar) signalProcess() (err error) {
 	return nil
 }
 
-func (s *Sidecar) checkProcessExit() {
+func (s *sidecar) checkProcessExit() {
```

Comment:
s.processRunning is used from multiple goroutines w/o any sort of lock

Reply:
Handled this with a mutex

Reply:
The mutex should be only held during each mutation of processRunning and not held across the call to s.process.Wait(), otherwise nobody will be able to take the lock while the process is alive.